File: frontend/src/api/mappers/Creator.mapper.ts
import dayjs from 'dayjs';

import { type ICreator } from '@/models/Creator.model';
import { type IApiCreator } from '@/api/api-models/ApiCreator.type';

export const creatorMapper = {

    map(creator: IApiCreator): ICreator {
        return {
            reference: creator.reference,
            createdAt: dayjs(creator.createdAt),
            name: creator.name,
            storeUrl: creator.storeUrl,
            description: creator.description,
        };
    },

};

File: backend/ClickStitch/Models/Mappers/CreatorMapper.cs
using Data.Records;

namespace ClickStitch.Models.Mappers;

public static class CreatorMapper
{
    public static CreatorModel Map(CreatorRecord creator) => new()
    {
        Reference = creator.Reference,
        CreatedAt = creator.CreatedAt,
        Name = creator.Name,
        StoreUrl = creator.StoreUrl,
        Description = creator.Description ?? ""
    };
}